coalib/bearlib/languages/definitions/Haskell.py

```diff
@@ -0,0 +1,19 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class Haskell:
+    extensions = '.hs',
+    versions = 1.0, 1.1, 1.2, 1.3, 1.4, 98, 2010
+    special_chars = list('!#$%&*+./<=>?@\\^|-~')
+    comment_delimiters = '--',
+    multiline_comment_delimiters = {'{-': '-}'}
+    keywords = [
+        '-<', '-<<', '::', ';', '<-', ',', '_', '{, }',
+        'case', 'of', 'class', 'infixl', 'foreign',
+        'data', 'data family', 'data instance', 'default',
+        'deriving', 'riving instance', 'do', 'forall',
+        'hiding', 'if', 'then', 'else', 'import', 'infix',
+        'infixr', 'instance', 'let', 'in', 'mdo', 'module',
+        'proc', 'qualified', 'rec', 'type', 'type family',
+        'type instance', 'where', 'newtype']
```

Comment:
Since `type` and `instance` already exists, I don't think you need to write it again as `type instance`. Also, where is `show` keyword?